### PR TITLE
Implement the home UI

### LINKED ISSUE #24: Implement the Home UI
[Figma design](https://www.figma.com/design/d60yqkXZspwUMpdiyyFyZn/GlowUp-application?node-id=229-3231&node-type=frame&t=TRiibIHSioJCNBzb-0)

### PR DESCRIPTION
**Pull request by thanhdo29**
## Related issues
Resolve #24 

## How to test
- Run app
- Change to BASE_URL in `HttpClient.ts` use IPv4

## Concern points
Runs on dark mode and light mode

### CODE REVIEW

**Comment by KMTus** [comment]
image
Can you make a banner like this?

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
There are so many bugs

Simulator Screenshot - iPhone 15 Pro Max - 2024-10-10 at 20 10 48

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
Simulator Screenshot - iPhone 16 Plus - 2024-10-12 at 08 31 03

**Comment by NguyenVanViet2004** [comment]
image

**Comment by thanhdo29** [comment]
> image



> image

I don't understand the View All part. Do you mean to display the selected Combo on the Banner?

**Comment by NguyenVanViet2004** [comment]
@thanhdo29  No, I want to redirect the user to a new template when the user clicks "View all" to show all combos

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
Call api to display combo

**Review by KMTus — APPROVED** [review]

File: components/templates/HomeTemplate.tsx [hunk 1]
```diff
@@ -1,36 +1,103 @@
...
+                color="black" />
+            </View>
+          </View>
+
+          <MemoizedBanner
+            marginTop={15}
+            img={require('~/assets/images/imgBanner.png')}
+            nameCombo="Morning Special!"
+            percent="20"
+          />
```
Comment (truongtdph40209) [inline components/templates/HomeTemplate.tsx:65]: Use api banner

File: components/templates/HomeTemplate.tsx [hunk 2]
```diff
@@ -1,36 +1,103 @@
...
+import LinearGradientBackground from '~/components/molecules/LinearGradientBackground'
+import ListCombo from '~/components/molecules/ListCombo'
+import SpecialComboCard from '~/components/molecules/SpecialComboCard'
+import getColors from '~/constants/Colors'
+import { RADIUS_BUTTON } from '~/constants/Constants'
+import useFetchCombo from '~/hooks/useFetchCombo'
+import useTranslation from '~/hooks/useTranslation'
+import type Combo from '~/interfaces/Combo'
+
+import Loading from '../atoms/Loading'
```
Comment (truongtdph40209) [inline components/templates/HomeTemplate.tsx:17]: Relative path

**Review by truongtdph40209 — COMMENTED** [review]
Fix notification icon for darkmode
image

File: apis/HttpClient.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ const headers = {
 export interface Response<T = any> {
   success: boolean
   message: string
-  result: T
+  data: T
```
Comment (truongtdph40209) [inline apis/HttpClient.ts:16]: image
Update inteface

File: hooks/useFetchBanner.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    const fetchCombos = async (): Promise<void> => {
+      try {
+        setIsLoading(true)
+        const response =
+        await request.get<Banner[]>('banner')
+        if (response?.success && !isNil(response.data)) {
+          setBanner(response.data)
```
Comment (truongtdph40209) [inline hooks/useFetchBanner.ts:23]: Sync with backend

File: hooks/useFetchCombo.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+const useFetchCombo = (): any => {
+  const [combos, setCombos] = useState<Combo[]>([])
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    const fetchCombos = async (): Promise<void> => {
+      try {
+        const response = await request.get<Combo[]>('combo')
+        if (response?.success && !isNil(response.data)) {
+          setCombos(response.data)
```
Comment (truongtdph40209) [inline hooks/useFetchCombo.ts:16]: Sync with backend data => result

**Review by truongtdph40209 — COMMENTED** [review]
You should pull the latest code from the backend

**Review by NguyenVanViet2004 — APPROVED** [review]

**Pull request merged**